tests/functional/test_prepare_sources.py

```diff
@@ -40,3 +43,32 @@ def test_prepare_sources_command(cwd_upstream_or_distgit):
     assert tarball_path.exists()
     specfile_path = next(result_dir.glob("*.spec"))
     assert specfile_path.exists()
+
+
+@pytest.mark.parametrize(
+    "cwd_upstream_or_distgit",
+    ["upstream", "upstream-with-multiple-sources"],  # Only test upstream variants
+    indirect=True,
+)
+def test_prepare_sources_with_preserve_spec(cwd_upstream_or_distgit):
+    """Test that --preserve-spec flag prevents spec file modifications"""
+    # Read original spec file content before prepare-sources
+    original_spec_path = next(cwd_upstream_or_distgit.glob("*.spec"))
+    original_spec = Specfile(original_spec_path)
+    original_version = original_spec.expanded_version
+    original_release = original_spec.expanded_release
+    with original_spec.sources() as sources:
+        original_source0 = sources[0].location
+
+    call_real_packit(
+        parameters=["--debug", "prepare-sources", "--preserve-spec"],
+        cwd=cwd_upstream_or_distgit,
+    )
+
+    # Spec file in original location should not be modified
+    # Version, Release, and Source0 should remain unchanged
+    spec_after = Specfile(original_spec_path)
```

Comment:
You could just call `original_spec.reload()` (probably after giving `original_spec` more suitable name).